src/main/java/com/mongodb/spark/sql/connector/config/ReadConfig.java

```diff
@@ -161,6 +161,66 @@ public final class ReadConfig extends AbstractMongoConfig {
 
   private static final boolean AGGREGATION_ALLOW_DISK_USE_DEFAULT = true;
 
+  /**
+   * Parse Mode using the existing Spark json parse mode configuration names
+   */
+  public enum ParseMode {
+    /**
+     * Fail if a data error occurs
+     */
+    FAILFAST,
+    /**
+     * Ignore any data errors for a field and replace with null
+     */
+    PERMISSIVE,
+    /**
+     * Drop any rows that contains a data error
+     */
+    DROPMALFORMED;
+
+    static ParseMode fromString(final String userParseMode) {
+      try {
```

Comment:
If `null` is provided through the `option()` method, it currently results in a `NullPointerException`. To make this more user-friendly and consistent, we could throw a `ConfigException` , which provides clearer feedback.

```suggestion
     validateConfig(userParseMode, Objects::nonNull, () -> "The userParseMode can't be null");
      try {
      
```

Reply:
I ended up reverting as this method is internal and only called via the readConfig method where there is a default value for the parse method.